src/status_im/chat/commands/impl/transactions.cljs

```diff
@@ -304,39 +300,21 @@
     (send-preview command-message))
   protocol/Yielding
   (yield-control [_ {{{amount :amount asset :asset} :params} :content} {:keys [db] :as cofx}]
-    ;; Prefill wallet and navigate there
-    (let [recipient-contact (or
-                             (get-in db [:contacts/contacts (:current-chat-id db)])
-                             (db.contact/public-key->new-contact (:current-chat-id db)))
-
-          sender-account        (:account/account db)
-          chain                 (keyword (:chain db))
-          symbol-param          (keyword asset)
-          all-tokens            (:wallet/all-tokens db)
-          {:keys [symbol decimals]} (tokens/asset-for all-tokens chain symbol-param)
-          {:keys [value error]}     (wallet.db/parse-amount amount decimals)
-          next-view-id              (if (:wallet-set-up-passed? sender-account)
-                                      :wallet-send-modal-stack
-                                      :wallet-send-modal-stack-with-onboarding)]
-      (fx/merge cofx
-                {:db (-> db
-                         (assoc-in [:navigation/screen-params :wallet-send-modal-stack :modal?] true)
-                         (update-in [:wallet :send-transaction]
-                                    assoc
-                                    :amount (money/formatted->internal value symbol decimals)
-                                    :amount-text amount
-                                    :amount-error error)
-                         (choose-recipient.events/fill-request-details
-                          (transaction-details recipient-contact symbol) false)
-                         (update-in [:wallet :send-transaction]
-                                    dissoc :id :password :wrong-password?))
-                 ;; TODO(janherich) - refactor wallet send events, updating gas price
-                 ;; is generic thing which shouldn't be defined in wallet.send, then
-                 ;; we can include the utility helper without running into circ-dep problem
-                 :wallet/update-gas-price
-                 {:success-event :wallet/update-gas-price-success
-                  :edit?         false}}
-                (navigation/navigate-to-cofx next-view-id {}))))
+    (let [{:keys [symbol decimals address]} (tokens/asset-for (:wallet/all-tokens db) (keyword (:chain db)) (keyword asset))
+          {:keys [value]} (wallet.db/parse-amount amount decimals)
+          current-chat-id (:current-chat-id db)
+          amount-hex      (str "0x" (abi-spec/number-to-hex (money/formatted->internal value symbol decimals)))
+          to              (ethereum/public-key->address current-chat-id)
+          to-norm         (ethereum/normalized-address (if (= symbol :ETH) to address))
+          tx-obj          (if (= symbol :ETH)
```

Comment:
I am not a big fan of this tx-obj everywhere, it isn't really defined anywhere what it is about compared to just transaction

Reply:
in the end this tx-obj is just (select-keys transaction [:to :from :data])

Reply:
i make tx-obj separate because in that case I'm sure it won't be changed and will be signed as is, if we mix it with a transaction for signing screen, it might be messed